src/mainTemplate.json

```diff
@@ -1321,7 +1322,12 @@
             "value": "[variables('networkSettings')]"
           },
           "azureCloudStorageAccount": {
-            "value": "[variables('azureCloudStorageAccount')]"
+            "value": {
```

Comment:
Want to keep the provisioning of a storage account for azure-repository plugin separate.

Reply:
Yeah, maybe that's better. But could we add the storageAccountSuffix support for non-globle azure environment.

The automation provisioning of the registry storageAccount may be an alternative option for user. What do you think?

Reply:
> But could we add the storageAccountSuffix support for non-globle azure environment.

I agree that we should allow a `storageAccountSuffix` parameter to be passed and configure the repository-azure plugin with it.

> The automation provisioning of the registry storageAccount may be an alternative option for user. What do you think?

My concerns with allowing this option are that 
1. it increases the complexity of the template
2. with the storage account configured within the same resource group, accidentally deleting all the snapshots along with the cluster seems like it would be a much easier mistake to make by e.g. deleting the resource group.

With the current implementation, a user can create a resource group with a storage account, then perform a template deployment into that resource group, or deploy into a separate resource group.

Interested to hear your thoughts on this @Mpdreamz.